lib/jekyll-seo-tag/drop.rb

```diff
@@ -0,0 +1,239 @@
+
+          author["twitter"] ||= author["name"]
+          author["twitter"].delete! "@"
+          author.to_liquid
+        end
+      end
+
+      def date_modified
+        @date_modified ||= begin
+          date = if page["seo"] && page["seo"]["date_modified"]
+                   page["seo"]["date_modified"]
+                 else
+                   page["last_modified_at"] || page["date"]
+                 end
+          filters.date_to_xmlschema(date) if date
+        end
+      end
+
+      def date_published
+        @date_published ||= filters.date_to_xmlschema(page["date"]) if page["date"]
+      end
+
+      def type
+        @type ||= begin
+          if page["seo"] && page["seo"]["type"]
+            page["seo"]["type"]
+          elsif homepage_or_about?
+            "WebSite"
+          elsif page["date"]
+            "BlogPosting"
+          else
+            "WebPage"
+          end
+        end
+      end
+
+      def links
+        @links ||= begin
+          if page["seo"] && page["seo"]["links"]
+            page["seo"]["links"]
+          elsif homepage_or_about? && site["social"] && site["social"]["links"]
+            site["social"]["links"]
+          end
+        end
+      end
+
+      def logo
+        @logo ||= begin
+          return unless site["logo"]
+          if absolute_url? site["logo"]
+            filters.uri_escape site["logo"]
+          else
+            filters.uri_escape filters.absolute_url site["logo"]
+          end
+        end
+      end
+
+      # Returns nil or a hash representing the page image
+      # The image hash will always contain a path, pulled from:
+      #
+      # 1. The `image` key if it's a string
+      # 2. The `image.path` key if it's a hash
+      # 3. The `image.facebook` key
+      # 4. The `image.twitter` key
+      #
+      # The resulting path is always an absolute URL
+      def image
+        return @image if defined?(@image)
+
+        image = page["image"]
+        return @image = nil unless image
+
+        image = { "path" => image } if image.is_a?(String)
+        image["path"] ||= image["facebook"] || image["twitter"]
+
+        unless absolute_url? image["path"]
+          image["path"] = filters.absolute_url image["path"]
+        end
+
+        image["path"] = filters.uri_escape image["path"]
+
+        @image = image.to_liquid
+      end
+
+      def page_lang
+        @page_lang ||= page["lang"] || site["lang"] || "en_US"
+      end
+
+      def canonical_url
+        @canonical_url ||= filters.absolute_url(page["url"]).gsub(%r!/index\.html$!, "/")
+      end
+
+      private
+
+      def filters
+        @filters ||= Jekyll::SeoTag::Filters.new(@context)
+      end
+
+      def page
+        @page ||= @context.registers[:page].to_liquid
+      end
+
+      def site
+        @site ||= @context.registers[:site].site_payload["site"].to_liquid
+      end
+
+      def homepage_or_about?
+        ["/", "/index.html", "/about/"].include? page["url"]
```

Comment:
Moved to regex and added tests via cfe7e56.

I'm not super excited to support `/index/` or `about.html`. (GitHub Pages lazy indexes and ugly permalinks). I could see a case for the second if there was strong desire.